README.md

```diff
@@ -0,0 +1,24 @@
+# daly
+Simple VM for a Dyon--subset
+
+To run use:
+
+    cargo run
+  
+To see what is happening internally, you can enable logging:
+
+    RUST_LOG=daly cargo run 
+
+## Current state
+
+* `main.rs` implements a simple interpreter for some dyon-bytecode. `main()` contains the bytecode of [this](https://github.com/greenMT/example-programs/blob/master/example-programs/dyon/min_loop.dyon) program.
+
+* the interpreter traces execution of loops
+
+* `tracerunner.rs` contains an independent execution engine for generated traces
+
+
+### Optimisations
+
+**Inlining**
+Inlining of function calls is performed. Necessary steps for deoptimisation can be found in `tracerunner::Runner::recover`.
```

Comment:
Let me give you a (somewhat artificial) example. Imagine you've got a function `f` which inspects the stack. If you inline `f`, you then have to deopt so that it can inspect the stack; you'd be better off not actually inlining it but you might only realise that *after* you've inlined it.

That said, in general, you're right. But I think the point Jasper is making is not about uninlining, so much as it's about "we called `f`, a guard failed, so we have to blackhole".